dynamicExecutable/CMakeLists.txt

```diff
@@ -1,32 +1,54 @@
-cmake_minimum_required(VERSION 3.0)
+cmake_minimum_required(VERSION 3.15)
```

Comment:
I am also just noticing that you bumped up the required cmake version to 3.15. I assume this is required by the GPU compiler, but this change makes it impossible to build Shasta on Ubuntu 16.04, which has cmake 3.5.1 (I don't want to force users to install a different cmake version from what comes with their system). Even with `BUILD_FOR_GPU` turned off, the build would fail on Ubuntu 16.04 because of this. 

Creating a separate cmake file for the gpu compilations may fix this, if cmake lets us specify a different minimum version in the gpu subdirectory. Another way would be to not use new cmake features, but this might not be possible for the gpu build.

In summary: ok to say that when `BUILD_FOR_GPU` is on, Ubuntu 18.04 is required. But not when `BUILD_FOR_GPU` is off. It's too early drop build support for Ubuntu 16.04